nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PostSlack.java

```diff
@@ -0,0 +1,516 @@
+            .build();
+
+    public static final AllowableValue UPLOAD_FLOWFILE_YES = new AllowableValue(
+            "true",
+            "Yes",
+            "Upload and attach FlowFile content to the Slack message."
+    );
+
+    public static final AllowableValue UPLOAD_FLOWFILE_NO = new AllowableValue(
+            "false",
+            "No",
+            "Don't upload and attach FlowFile content to the Slack message."
+    );
+
+    public static final PropertyDescriptor UPLOAD_FLOWFILE = new PropertyDescriptor.Builder()
+            .name("upload-flowfile")
+            .displayName("Upload FlowFile")
+            .description("Whether or not to upload and attach the FlowFile content to the Slack message.")
+            .allowableValues(UPLOAD_FLOWFILE_YES, UPLOAD_FLOWFILE_NO)
+            .required(true)
+            .defaultValue("false")
+            .build();
+
+    public static final PropertyDescriptor FILE_TITLE = new PropertyDescriptor.Builder()
+            .name("file-title")
+            .displayName("File Title")
+            .description("Title of the file displayed in the Slack message." +
+                    " The property value will only be used if 'Upload FlowFile' has been set to 'Yes'.")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor FILE_NAME = new PropertyDescriptor.Builder()
+            .name("file-name")
+            .displayName("File Name")
+            .description("Name of the file to be uploaded." +
+                    " The property value will only be used if 'Upload FlowFile' has been set to 'Yes'." +
+                    " If the property evaluated to null or empty string, then the file name will be set to 'file' in the Slack message.")
+            .defaultValue("${" + CoreAttributes.FILENAME.key() + "}")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor FILE_MIME_TYPE = new PropertyDescriptor.Builder()
+            .name("file-mime-type")
+            .displayName("File Mime Type")
+            .description("Mime type of the file to be uploaded." +
+                    " The property value will only be used if 'Upload FlowFile' has been set to 'Yes'." +
+                    " If the property evaluated to null or empty string, then the mime type will be set to 'application/octet-stream' in the Slack message.")
+            .defaultValue("${" + CoreAttributes.MIME_TYPE.key() + "}")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor
```

Comment:
By default (when the username is not specified) the displayed name in the message is the Slack App name of the bot user who sent the message. I think this should be used always. 

I just added the username property because PutSlack has the same parameter (however, it does not work in PutSlack, because PutSlack uses Slack webhooks and the new version of Slack webhooks does not support the username parameter any more, I guess for the same reason you mentioned).

Just before opening the PR, I was thinking about removing the username/icon properties at all, but in the end I left them in the code to see what the reviewers think about it.

Now I would definitely vote for removing it from the processor properties. What is your opinion?

Another reason is that username parameter works only when you send a text message but not for file upload messages. File uploads always use the Slack app name, because file upload API call does not have the username parameter. This would not be a really consistent behaviour for the end user.

Reply:
The inconsistency seems to belong to the Slack API, right? In that case it's not really our fault, and I think as long as it is documented for the processor then we'd be ok. That being said, it would be nice to have a fake username that is unique to the instance/thread that calls it, so we can trace it back to the logs, but not allow for spoofing.

Reply:
In order to prevent spoofing, we need to remove the username as a configurable property by the end user.

Regarding the instance/thread id logging: do you mean our nifi logs or also sending this id to Slack as the username?
In the second case this long technical id would be displayed in Slack which not really user friendly I think. I would prefer to see in Slack that "Whatever App/Bot" sent me a message rather than an id. So I would use the App name for this (which is the default).

Reply:
Yeah I agree it would be really long, was just thinking of a way to have it be unique to the instance/thread so you knew which one was posting, in case one went nutz and starting posting a million times 😛 I guess a single readable name is probably better in the general case.